src/transforms/route.rs

```diff
@@ -16,48 +16,127 @@ use crate::{
 
 //------------------------------------------------------------------------------
 
+fn build_conditions(
+    config: &RouteConfig,
+    context: &TransformContext,
+) -> crate::Result<Vec<(String, Condition)>> {
+    let mut conditions = Vec::with_capacity(config.route.len());
+    for (output_name, condition) in config.route.iter() {
+        let condition = condition.build(&context.enrichment_tables)?;
+        conditions.push((output_name.clone(), condition));
+    }
+    Ok(conditions)
```

Comment:
This pattern looks a lot like a map with the final `collect` transposing the `Result`. Maybe something like this?
```suggestion
    config.route
        .iter()
        .map(|(output_name, condition)| {
            condition.build(&context.enrichment_tables).map(|condition| (output_name.clone(), condition))
        })
        .collect()
```